src/test/java/com/arpnetworking/test/TelemetryClient.java

```diff
@@ -0,0 +1,276 @@
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.Dsl;
+import org.asynchttpclient.ws.WebSocket;
+import org.asynchttpclient.ws.WebSocketListener;
+import org.asynchttpclient.ws.WebSocketUpgradeHandler;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+/**
+ * Class which consumes streaming telemetry from MAD.
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot io)
+ */
+public final class TelemetryClient {
+
+    /**
+     * Retrieve the singleton instance of the {@code TelemetryClient}.
+     *
+     * @return the singleton instance of the {@code TelemetryClient}
+     */
+    public static synchronized TelemetryClient getInstance() {
+        if (gTelemetryClient == null) {
+            try {
+                gTelemetryClient = new TelemetryClient(
+                        getEnvOrDefault("MAD_HOST", "localhost"),
+                        getEnvOrDefault("MAD_HTTP_PORT", DEFAULT_MAD_HTTP_PORT, Integer::parseInt));
+                // CHECKSTYLE.OFF: IllegalCatch - Catastrophic failure
+            } catch (final ExecutionException | InterruptedException e) {
+                // CHECKSTYLE.ON: IllegalCatch
+                throw new RuntimeException(e);
+            }
+        }
+        return gTelemetryClient;
+    }
+
+    /**
+     * Start receiving data for the specified time series.
+     *
+     * @param service the service name
+     * @param metricName the metric name
+     * @param statistic the statistic
+     * @param callback the consumer to invoke with the received value(s)
+     */
+    public void subscribe(
+            final String service,
+            final String metricName,
+            final Statistic statistic,
+            final Consumer<Double> callback) {
+        try {
+            _webSocket.sendTextFrame(
+                    OBJECT_MAPPER.writeValueAsString(
+                            ImmutableMap.of(
+                                    "command", "subscribeMetric",
+                                    "service", service,
+                                    "metric", metricName,
+                                    "statistic", statistic.getName())));
+            _callbacks.put(String.format("%s/%s/%s", service, metricName, statistic.getName()), callback);
```

Comment:
Consider refactoring `String.format("%s/%s/%s", service, metricName, statisticName)` into a static method to force consistency?

Reply:
Done!